### PR TITLE
Migrate to xunit

### PR DESCRIPTION
**Pull request by hemdagem**
Migrated to Xunit
Removed NUnit
Removed Moq

### CODE REVIEW

File: src/Noobot.Tests.Unit/packages.config [hunk 1]
```diff
@@ -70,4 +68,12 @@
...
   <package id="System.Xml.XDocument" version="4.3.0" targetFramework="net461" />
   <package id="WebsocketClientLite.PCL" version="4.0.6" targetFramework="net461" />
+  <package id="xunit" version="2.2.0" targetFramework="net461" />
+  <package id="xunit.abstractions" version="2.0.1" targetFramework="net461" />
+  <package id="xunit.assert" version="2.2.0" targetFramework="net461" />
+  <package id="xunit.core" version="2.2.0" targetFramework="net461" />
+  <package id="xunit.extensibility.core" version="2.2.0" targetFramework="net461" />
+  <package id="xunit.extensibility.execution" version="2.2.0" targetFramework="net461" />
+  <package id="xunit.runner.visualstudio" version="2.2.0" targetFramework="net461" developmentDependency="true" />
+  <package id="xunit.should" version="1.1" targetFramework="net461" />
```
Comment (Workshop2) [inline src/Noobot.Tests.Unit/packages.config]: Careful using this version. I had problems as it uses an older xunit. Can this be replaced with `Xunit.Should.Extension`

**Review by Workshop2 — CHANGES_REQUESTED** [review]
Are you able to swap the should library that is used

File: src/Noobot.Tests.Unit/packages.config (hunk 1 above)
Comment (hemdagem) [reply src/Noobot.Tests.Unit/packages.config]: @Workshop2 made that change and pushed again.

**Review by Workshop2 — APPROVED** [review]

**Comment by Workshop2** [comment]
Fantastic work @hemdagem

**Pull request merged**